### PR TITLE
Hw 5 read file

### PR DESCRIPTION
**Pull request by IhorRome**

### CODE REVIEW

File: src/main/java/core/basesyntax/FileWork.java [hunk 1]
```diff
@@ -1,7 +1,63 @@
...
+                        } while (notEndOfWord(value));
+                    } else {
+                        do {
+                            value = reader.read();
+                        } while (notEndOfWord(value));
+                    }
+                } else {
+                    value = reader.read();
+                }
+            }
```
Comment (Resci) [inline src/main/java/core/basesyntax/FileWork.java:35]: Too complex if-else constructions, let's simplify

**Review by Resci — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FileWork.java [hunk 2]
```diff
@@ -1,7 +1,70 @@
...
+
 public class FileWork {
+    private static final char SPECIFIED_CHARACTER = 'w';
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            int value = reader.read();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:16]: let's use `readLine()` instead - what do you think?

File: src/main/java/core/basesyntax/FileWork.java [hunk 3]
```diff
@@ -1,7 +1,70 @@
...
+            while (value != -1) {
+                if (Character.toLowerCase((char)value) == SPECIFIED_CHARACTER) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:18]: use startsWith() instead

File: src/main/java/core/basesyntax/FileWork.java [hunk 4]
```diff
@@ -1,7 +1,70 @@
...
+                && (char) value != ','
+                && (char) value != '-'
+                && (char) value != '!'
+                && (char) value != '?'
+                && (char) value != '\n'
+                && (char) value != '\r'
+                && value != -1;
+    }
+
+    public String [] toStringArray(StringBuilder builder) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:61]: let's make the helper methods private

File: src/main/java/core/basesyntax/FileWork.java [hunk 5]
```diff
@@ -1,7 +1,70 @@
...
+        if (builder.length() != 0) {
+            String [] result = builder.toString().split(" ");
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:63]:
```suggestion
            String[] result = builder.toString().split(" ");
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 6]
```diff
@@ -1,7 +1,70 @@
...
+            Arrays.sort(result);
+            return result;
+        } else {
+            return new String[] {};
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/FileWork.java:67]:
```suggestion
            return new String[0];
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FileWork.java [hunk 7]
```diff
@@ -1,7 +1,51 @@
...
+        File file = new File(fileName);
+        StringBuilder builder = new StringBuilder();
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String value = reader.readLine();
+            while (value != null) {
+                getAllWordsWithW(value, builder);
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("can't read from file");
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java:23]:
```suggestion
            throw new RuntimeException("can't read from file", e);
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 8]
```diff
@@ -1,7 +1,51 @@
...
+        }
+        return toStringArray(builder);
+    }
+
+    private void getAllWordsWithW(String input, StringBuilder builder) {
+        input = input.replaceAll("[^a-z A-Z]", "")
+                .toLowerCase(Locale.ROOT);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java:30]: we cam simplify to toLowerCase()

File: src/main/java/core/basesyntax/FileWork.java [hunk 9]
```diff
@@ -1,7 +1,51 @@
...
+        String[] wordsFromLine = input.split(" ");
+        for (String word : wordsFromLine) {
+            if (word.startsWith(SPECIFIED_CHARACTER)) {
+                if (builder.length() != 0) {
+                    builder.append(' ');
+                }
+                builder.append(word);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java:37]:
```suggestion
                builder.append(word).append(' ');
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 10]
```diff
@@ -1,7 +1,51 @@
...
+            }
+        }
+    }
+
+    private String[] toStringArray(StringBuilder builder) {
+        if (builder.length() != 0) {
+            String[] result = builder.toString().split(" ");
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java:44]:
```suggestion
            String[] result = builder.toString().trim().split(" ");
```

File: src/main/java/core/basesyntax/FileWork.java [hunk 11]
```diff
@@ -1,7 +1,51 @@
...
+            Arrays.sort(result);
+            return result;
+        } else {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/FileWork.java:47]: do we need else after return statement?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**